app/controllers/candidate_interface/gcse/science/grade_controller.rb

```diff
@@ -1,39 +1,65 @@
 module CandidateInterface
   class Gcse::Science::GradeController < CandidateInterfaceController
     include Gcse::GradeControllerConcern
-
     before_action :redirect_to_dashboard_if_submitted
     before_action :set_subject
 
     def edit
-      @application_qualification = details_form
-      @qualification_type = details_form.qualification.qualification_type
+      @gcse_grade_form = science_gcse_grade_form
+      @qualification_type = gcse_science_qualification.qualification_type
+
+      render view_path
     end
 
     def update
-      @qualification_type = details_form.qualification.qualification_type
+      @gcse_grade_form = science_gcse_grade_form.assign_values(science_details_params)
 
-      details_form.grade = details_params[:grade]
-      details_form.other_grade = details_params[:other_grade]
+      if @gcse_grade_form.save_grades
+        update_gcse_completed(false)
+        redirect_to next_path
+      else
+        track_validation_error(@gcse_grade_form)
+        render view_path
+      end
+    end
 
-      @application_qualification = details_form.save_grade
+  private
 
-      if @application_qualification
-        update_gcse_completed(false)
-        redirect_to next_gcse_path
+    def view_path
+      if gcse_qualification? && FeatureFlag.active?(:science_gcse_awards)
+        'candidate_interface/gcse/science/grade/awards_edit'
       else
-        @application_qualification = details_form
-        track_validation_error(@application_qualification)
+        'candidate_interface/gcse/science/grade/edit'
```

Comment:
Any reason not to use path helpers here?

Reply:
These are just the paths to the appropriate view (not redirecting to a url path)

Reply:
ah, skimming too quickly! :+1: